File: Algorithmic_Toolbox/week2_algorithmic_warmup/my_fib1.py
# python3
import sys
from random import randint
import time
'''
Quickly calculate the a Fibonnaci sequence
'''
'''
insert base solution here
'''
def test_solution(n):
        if (n <= 1):
            return n

        return test_solution(n - 1) + test_solution(n - 2)


'''
code for comparing the base solution and a new solution
'''
def stress_test():
    while True:
        input = generate()

        base_start = time.time()
        base_solve = test_solution(input)
        base_end = time.time()

        new_start = time.time()
        new_solve = solution(input)
        new_end = time.time()

        print (input)
        print ('Base: ', base_solve, 'New: ', new_solve)
        print ('Base: ', base_end-base_start, 'New: ', new_end-new_start)
        if base_solve != new_solve:
            break
        else:
            print ('Pass')

'''
generator for input to both solutions
'''
def generate():
    n = randint(2, 15)

    return n


'''
new solution to be tested
'''
def solution(input):
    if input <= 1:
        return input
    cache = [0 for _ in range(input)]
    cache[0] = 1
    for i in range(1,len(cache)):
        cache[i] = cache[i-1] + cache[i-2]


    return cache[-1]


if __name__ == '__main__':

    ''' manual test '''
    # input = generate()
    # print (input)
    # input = int(input())
    # start = time.time()
    # print (solution(input))
    # end = time.time()
    # print (end-start)

    '''stress test'''
    # stress_test()

    ''' submission '''
    input = int(input())
    print (solution(input))

File: Algorithmic_Toolbox/week2_algorithmic_warmup/my_fib1.cpp
#include <iostream>
#include <cassert>
#include <random>
/*
create a function that calculates the Fibonacci number
*/

/*
Base solution, naive but works
*/
int base_solution(int n) {
    if (n <= 1)
        return n;

    return base_solution(n - 1) + base_solution(n - 2);
}

/*
New, fast solution to test
*/
int test_solution(int n) {
    // write your code here

    return 0;
}

/*
Generator function for stress testing
*/
void generate(){
  int input = rand() % 10 + 1;
  return input
}


// void stress_test() {
//   while True {
//     n = generate()
//
//   }
//     assert(fibonacci_fast(3) == 2);
//     assert(fibonacci_fast(10) == 55);
//     for (int n = 0; n < 20; ++n)
//         assert(fibonacci_fast(n) == fibonacci_naive(n));
// }


/*
call functions
*/
int main() {
    // int n = 0;
    // std::cin >> n;
    while(True){
      std::cout << generate();

    }
    // std::cout << fibonacci_naive(n) << '\n';
    // // test_solution();
    // //std::cout << fibonacci_fast(n) << '\n';
    // return 0;
}
